tests/test_web.py

```
@@ -131,3 +135,152 @@ def test_validate_openid_identifier_rejects(identifier: str) -> None:
+        "not-the-token-we-issued",
+        "— not even ascii",
+    ],
+)
+def test_google_login_rejects_bad_state(
+    app: Flask, monkeypatch: pytest.MonkeyPatch, state: str | None
+) -> None:
+    calls = spy_on_token_exchange(monkeypatch)
+    client = app.test_client()
+
+    with client.session_transaction() as flask_session:
+        flask_session["google_login_token"] = "the-token-we-issued"
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "google_login_token"

(S105)